weld.fs.js

```diff
@@ -142,29 +151,44 @@ export const weld = defineFeature(function(context is Context, id is Id, definit
                 definition.filletCornerShape is FilletCornerShape;
             }
 
-            else if (definition.weldType == WeldType.V_BUTT_WELD || definition.weldType == WeldType.DOUBLE_V_BUTT_WELD)
+            else
             {
                 annotation { "Name" : "Welded edge", "Filter" : EntityType.EDGE, "MaxNumberOfPicks" : 1 }
-                definition.vButtEdge is Query;
+                definition.buttEdge is Query;
 
                 annotation { "Name" : "Shape", "UIHint" : ["REMEMBER_PREVIOUS_VALUE", "SHOW_LABEL"] }
-                definition.vButtShape is VButtShape;
+                definition.buttShape is ButtShape;
 
-                annotation { "Name" : "Angle", "UIHint" : "REMEMBER_PREVIOUS_VALUE" }
-                isAngle(definition.vButtAngle, ANGLE_STRICT_180_BOUNDS);
-
-                annotation { "Name" : "Root Gap", "UIHint" : "REMEMBER_PREVIOUS_VALUE", "Default" : true }
-                definition.vButtRootGap is boolean;
+                if (definition.weldType == WeldType.SQUARE_BUTT_WELD)
+                {
+                    annotation { "Name" : "Distance", "UIHint" : "REMEMBER_PREVIOUS_VALUE" }
+                    isLength(definition.buttDist, BLEND_BOUNDS);
+                }
 
-                if (definition.vButtRootGap)
+                if (definition.weldType == WeldType.V_BUTT_WELD || definition.weldType == WeldType.DOUBLE_V_BUTT_WELD)
                 {
-                    annotation { "Name" : "Root width", "UIHint" : "REMEMBER_PREVIOUS_VALUE" }
-                    isLength(definition.vButtRootGapWidth, SHELL_OFFSET_BOUNDS);
+                    annotation { "Name" : "Angle", "UIHint" : "REMEMBER_PREVIOUS_VALUE" }
+                    isAngle(definition.buttAngle, ANGLE_STRICT_180_BOUNDS);
+                }
 
-                    annotation { "Name" : "Root height", "UIHint" : "REMEMBER_PREVIOUS_VALUE" }
-                    isLength(definition.vButtRootGapHeight, SHELL_OFFSET_BOUNDS);
+                if (definition.weldType == WeldType.U_BUTT_WELD || definition.weldType == WeldType.J_BUTT_WELD)
+                {
+                    annotation { "Name" : "Radius", "UIHint" : "REMEMBER_PREVIOUS_VALUE" }
+                    isLength(definition.buttRadius, BLEND_BOUNDS);
                 }
             }
+
+            annotation { "Name" : "Root Gap", "UIHint" : "REMEMBER_PREVIOUS_VALUE", "Default" : true }
+            definition.buttRootGap is boolean;
+
+            if (definition.buttRootGap)
+            {
+                annotation { "Name" : "Root width", "UIHint" : "REMEMBER_PREVIOUS_VALUE" }
+                isLength(definition.buttRootGapWidth, SHELL_OFFSET_BOUNDS);
+
+                annotation { "Name" : "Root height", "UIHint" : "REMEMBER_PREVIOUS_VALUE" }
+                isLength(definition.buttRootGapHeight, SHELL_OFFSET_BOUNDS);
+            }
```

Comment:
This section for root gap should be moved up and put in an `if` like this:
```
if(definition.weldType != WeldType.SQUARE_BUTT_WELD){
            annotation { "Name" : "Root Gap", "UIHint" : "REMEMBER_PREVIOUS_VALUE", "Default" : true }
            definition.buttRootGap is boolean;
             if (definition.buttRootGap)
            {
                annotation { "Name" : "Root width", "UIHint" : "REMEMBER_PREVIOUS_VALUE" }
                isLength(definition.buttRootGapWidth, SHELL_OFFSET_BOUNDS);
                 annotation { "Name" : "Root height", "UIHint" : "REMEMBER_PREVIOUS_VALUE" }
                isLength(definition.buttRootGapHeight, SHELL_OFFSET_BOUNDS);
            }
}
```

Reply:
Ok
You should still move it up, just without the if.
Currently it will show with a fillet weld